src/directives.ts

```diff
@@ -0,0 +1,136 @@
+      const scrollableContent = getClosest(el, '.scrollable-content');
+      if (scrollable && scrollableContent) {
+        const height = scrollableContent.offsetHeight;
+        setTimeout((): void => {
+          // if scrollableContent height is reduced in half second
+          // since an input got focus we assume soft keyboard is showing.
+          if (height > scrollableContent.offsetHeight) {
+            scrollTo(target as HTMLElement, scrollable, scrollableContent, 20);
+          }
+        }, 500);
+      }
+    };
+
+    target.addEventListener('focus', handler);
+    el.$destroy = (): void => {
+      target.removeEventListener('focus', handler);
+    };
+  },
+  unbind(el: DestroyHTMLElement): void {
+    el.$destroy();
+  },
+};
+
+export interface TrackClickDataAttributes {
+  'event-name'?: string;
+  'event-props'?: {[key: string]: any}, // eslint-disable-line @propelinc/no-explicit-any
+}
+
+export type trackClickHandlerFunction = (dataAttributes: TrackClickDataAttributes) => void;
+
+interface TrackClickDirectiveOptions extends DirectiveOptions {
+  handler: trackClickHandlerFunction;
+}
+
+/**
+ * Track an event to amplitude on click.
+ *
+ * Example usage:
+ * <div v-track-click event-name="foo" :event-props="{ bar: 'tzar' }">Click me</div>
+ */
+const trackClick: TrackClickDirectiveOptions = {
+  handler() {
+    return;
+  },
+  bind(el: DestroyHTMLElement, binding: DirectiveBinding, vnode: VNode): void {
+    const attrs = vnode.data && vnode.data.attrs ? vnode.data.attrs: {};
+
+    const _this = this;
+    function wrappedHandler(): void {
+      _this.handler(attrs);
+    }
+    el.addEventListener('click', wrappedHandler);
+    el.$destroy = (): void => el.removeEventListener('click', wrappedHandler);
+  },
+  unbind: (el: DestroyHTMLElement): void => {
+    // unbind is subject to a very specific race condition:
+    //
+    // If you set up both a v-track-click directive and an @click handler you can run into some issues
+    // The @click handler seems to get registered and to fire first.
+    // If that @click handler moves you away from the current page immediately, the unbind call will run
+    // before the click handler function setup on bind.
+    // This will lead to the click handler being removed and the handler not being run
+    //
+    // To fix that issue, we are using a setTimeout of 0. This yeilds the javascript event loop during the
+    // navigation away from the elements page. This allows the other click handlers to run before the
+    // event listeners are removed
+    setTimeout((): void => {
+      el.$destroy();
+    }, 0);
+  },
+};
+
+export function addDirectives(Vue: VueConstructor, trackClickHandler: trackClickHandlerFunction): void {
+  Vue.directive('infinite-scroll', infiniteScroll);
+  Vue.directive('scroll-on-focus', scrollOnFocus);
+
+  trackClick.handler = trackClickHandler;
+  Vue.directive('track-click', trackClick);
```

Comment:
Notice that there is a new pattern for defining the directives. Instead of defining them directly on the imported Vue singleton, I now defined three interfaces and then add them to the passed in Vue constructor on `addDirectives`

Reply:
love it.